lua/effects/thruster_sperm.lua

```diff
@@ -1,66 +1,36 @@
+function EFFECT:Init(data)
+	self:SetCollisionBounds(Vector(-16, -16, -16), Vector(16, 16, 16))
 
-EFFECT.Mat = Material( "thrusteraddon/sperm" )
-
-/*---------------------------------------------------------
-   Initializes the effect. The data is a table of data
-   which was passed from the server.
----------------------------------------------------------*/
-function EFFECT:Init( data )
-
-	local size = 16
-	self:SetCollisionBounds( Vector( -size,-size,-size ), Vector( size,size,size ) )
-
-	local Pos = data:GetOrigin() + data:GetNormal() * 2
-
-	self:SetPos( Pos )
-	self:SetAngles( data:GetNormal():Angle() + Angle( 0.01, 0.01, 0.01 ) )
-
-	self.Pos = data:GetOrigin()
-	self.Normal = data:GetNormal()
+	local origin, normal = data:GetOrigin(), data:GetNormal()
+	self:SetPos(origin + normal * 2)
+	self:SetAngles(normal:Angle())
```

Comment:
The random variation from before is missing. Also angles shouldn't be used at all.